docs/.vitepress/config.ts

```diff
@@ -129,161 +114,695 @@ export default defineConfig({
-              link: '/cli/change-detection/#module-change-detection',
-            },
-            {
-              text: 'Arbitrary Files',
-              link: '/cli/change-detection/#arbitrary-files-change-detection',
-            },
+            { text: 'Overview', link: '/cli/orchestration/' },
+            { text: 'Run Commands in Stacks', link: '/cli/orchestration/run-commands-in-stacks' },
+            { text: 'Terramate Scripts', link: '/cli/orchestration/scripts' },
+            { text: 'Tag Filter', link: '/cli/orchestration/tag-filter' },
+            { text: 'Runtime Configuration', link: '/cli/orchestration/runtime-configuration' },
+            { text: 'Safe Guards', link: '/cli/orchestration/safeguards' },
           ],
         },
         {
-          text: '🔗 Sharing Data',
+          text: '�️ Change Detection',
           collapsed: false,
           items: [
-            { text: 'Overview', link: '/cli/data-sharing/' },
-            { text: 'Globals', link: '/cli/data-sharing/globals' },
-            { text: 'Metadata', link: '/cli/data-sharing/metadata' },
-            { text: 'Map', link: '/cli/map' },
+            { text: 'Overview', link: '/cli/change-detection/' },
+            {
+              text: 'Integrations',
+              collapsed: true,
+              items: [
+                { text: 'Git Integration', link: '/cli/change-detection/integrations/git' },
+                { text: 'Terraform Integration', link: '/cli/change-detection/integrations/terraform' },
+                { text: 'OpenTofu Integration', link: '/cli/change-detection/integrations/opentofu' },
+                { text: 'Terragrunt Integration', link: '/cli/change-detection/integrations/terragrunt' },
+              ],
+            },
+            { text: 'File Watchers', link: '/cli/change-detection/file-watchers' },
           ],
         },
         {
-          text: '😍 Code Generation',
+          text: '🔗 Code Generation',
           collapsed: false,
           items: [
             { text: 'Overview', link: '/cli/code-generation/' },
             { text: 'Generate HCL', link: '/cli/code-generation/generate-hcl' },
             { text: 'Generate File', link: '/cli/code-generation/generate-file' },
-          ],
-        },
-        {
-          text: '🔧 Functions',
-          collapsed: false,
-          items: [
             {
-              text: 'Overview',
-              link: '/cli/functions/index',
+              text: 'Variables',
+              collapsed: true,
+              items: [
+                { text: 'Overview', link: '/cli/code-generation/variables/' },
+                { text: 'Globals', link: '/cli/code-generation/variables/globals' },
+                { text: 'Lets', link: '/cli/code-generation/variables/lets' },
+                { text: 'Metadata', link: '/cli/code-generation/variables/metadata' },
+                { text: 'Map', link: '/cli/code-generation/variables/map' },
+              ],
```

Comment:
The *Variables* nav should have not being moved into the *code generation* menu. The variables can be used in many features, not only code generation.

Reply:
This was explicitly requested by @mariux . @mariux are you ok with using moving this back as a top level point

Reply:
The `Lets` is the only variable only supported by code generation blocks. The others can be used whenever an expression is expected:

Example:
```
terramate run --eval echo "${terramate.stack.name}"
```

```
globals {
   map "varname" {
      # ...
   }
}
```

```
terramate run --eval global.varname
```

etc

Reply:
Additionally, the `script` block supports variables and it's an orchestration only feature. We even plan to add `lets` block to the `script` block.

Reply:
Yes, we will reference the features where needed, the main purpose is on code generation, the other features just support access to code gen capabilities here to improve the UX.